pkg/argo_client/kustomize.go

```diff
@@ -34,7 +35,11 @@ func addFile(repoRoot string, tempDir string, relPath string) error {
 	if err != nil {
 		return err
 	}
-	defer r.Close() // ignore error: file was opened read-only.
+	defer func() {
+		if err := r.Close(); err != nil {
+			log.Error().Err(err).Msg("failed to close file")
+		}
+	}() // ignore error: file was opened read-only.
```

Comment:
might be cleaner to wrap this in a function, since we're doing a very similar thing over and over:

```golang
func reportErrors(close func() error, msg string)  {
	if err := close(); err != nil {
		log.Error().Err(err).Msg(msg)
	}
}

// usage
defer reportErrors(r.Close, "failed to close file") 
```